src/components/tools/toolbar/ToolsDotMenu.js

```diff
@@ -61,16 +65,57 @@ function DotMenuItems(props) {
                 />
             )}
         </Hidden>,
+        <MenuItem
+            key={build(baseId, ids.MANAGE_TOOLS.ADD_TOOL_MI)}
+            id={build(baseId, ids.MANAGE_TOOLS.ADD_TOOL_MI)}
+            onClick={() => {
+                onClose();
+                onAddToolSelected();
+            }}
+        >
+            <ListItemIcon>
+                <Add fontSize="small" />
+            </ListItemIcon>
+            <ListItemText primary={t("addTool")} />
+        </MenuItem>,
+        allowEditing && (
+            <MenuItem
+                key={build(baseId, ids.MANAGE_TOOLS.EDIT_TOOL_MI)}
+                id={build(baseId, ids.MANAGE_TOOLS.EDIT_TOOL_MI)}
+                onClick={() => {
+                    onClose();
+                    onEditToolSelected();
+                }}
+            >
+                <ListItemIcon>
+                    <Edit fontSize="small" />
+                </ListItemIcon>
+                <ListItemText primary={t("edit")} />
+            </MenuItem>
+        ),
     ];
 }
 
 export default function ToolsDotMenu({
     ButtonProps,
     canShare,
     setSharingDlgOpen,
+    onEditToolSelected,
+    getSelectedTools,
     ...props
 }) {
-    const { baseId, isSingleSelection, onDetailsSelected } = props;
+    const {
+        baseId,
+        isSingleSelection,
+        onDetailsSelected,
+        onAddToolSelected,
+    } = props;
+    const selectedTools = getSelectedTools ? getSelectedTools() : null;
+    const allowEditing =
+        isSingleSelection &&
+        selectedTools &&
+        selectedTools.length > 0 &&
+        permissionHierarchy(selectedTools[0].permission) > 1;
```

Comment:
I am not sure if we should have a single function to test permission levels for different types of things (data, tools, apps etc...) even if they have currently share the same logic as of now(if thats what you are suggesting). In the future, changing this logic for Data for example might causes unnecessary side effect on Tools.

Reply:
I didn't think it would be a problem, since everything else shares the same permissions service, which is also modeled after our data store permissions model; but if it's still a concern and we don't want common permissions checks for everything, then at least those data permission util functions should be copied into the `components/tools/utils` namespace and used here (`permissionHierarchy > 1` doesn't really mean anything to me).

Reply:
Am little confused. Are you suggesting to copy `permissionHierarchy` over to `components/tools/utils` or having a function in `components/tools/utils` that check if the tool can be edited ? I was going to do the second.

Reply:
I was originally suggesting refactoring these functions into a common utils namespace:
https://github.com/cyverse-de/sonora/blob/36fde9118b2f50a4820b84aff62692f7e85f47b2/src/components/data/utils.js#L66-L91

If we don't want those as common permissions checks, then at least `isWritable` should be copied into `components/tools/utils` and that can be used here.